### DISCUSSION TITLE
More work for Mechanics: Vents, Ducts, Loaders and Doors.

### DISCUSSION THREAD

**Discussion by BlackMoons** [FakeFishGames/Barotrauma#13732] [Ideas and suggestions]
Mechanics have very little to do on a ship that isn't sinking/full of holes. The only mechanical objects that need constant repair is the O2 generator and engines (Vs 10~30 devices on a ship for an electrical engineer). This means they get very little job related EXP and generally most of the crew tends to help with welding the hull, further cutting into their EXP/skill gain. 

Pumps on basically all ship are designed to shut off when not needed and ballast pumps don't decay at 0% flow.

So how about giving them some more things to fix? Vents could decay when the linked O2 generator is on (In one mod, immersive repairs, they only decay when underwater, that could be cool too)

Doors and ducts could decay, based on one open or closing = 1 second worth of decay. Below 50% they could (have a chance to?) jam open/closed until repaired/crowbared into the opposite state, or if you want something more bot friendly, only jam open or just slow the opening/closing speed (Along with some grinding noise as it opens/closes that gets worse the lower the door condition). leaking like hulls do at lower condition would be cool too. 

Ammo loaders could decay, based on the assumption that each round fired = Weapon reload time worth of wear (so railguns get 5 seconds of wear per shot). 

Turrets could also honestly decay and take damage when firing (or even when just manned, though much slower), it would also be so nice to be able to shoot an enemy ships turrets to get them to stop firing instead of having to randomly aim through the ship in hopes to flood the gunnery/supercapacitor/electrical compartments. 

And yes, this means that sometimes your own turrets get damaged and your left defenseless and die.... But in Barotrauma, that is just one of MANY ways you can be left defenseless: Running out of ammo, damaged/discharged supercapacitors, broken electrical grid, broken loader, etc etc) Its not exactly anything new for horrible things to happen as a result of damage/system failure and that is just part of the fun of the game.

**Comment by bookslayer10** [comment]
I agree that Mechanic should get more to do, I think more mechanical devices wearing down over time would likely help that. I like the idea of doors having less of a binary damage system - giving them some in-between states of damage rather than just "Fixed" or "Broken." Same with all devices, really - have pumps get less effective as they take damage, have turret bases nerf firing speed as they break, have the oxygen generator generate less oxygen as it wears down, etc.

When it comes to turrets themselves taking damage, I really like the idea, but it feels like it could easily be way too dangerous and unfair. They should probably not break easily, as it would be extremely hard to repair them in the middle of a fight compared to most other submarine systems. Needing maintenance is different from needing to fix them when mudraptors are attacking.

Alternatively, they could break more easily, but never fully stop working - just have the fire rate and turning speed be significantly reduced, and maybe have strong attacks temporarily disable them.
